File: rumo9s.rb
resp = []

while gets()
    nr = $_.to_i
    ct = (nr == 9) ? 1 : 0

    while nr > 9
        nr = nr.to_s.split('').inject(0) {|s, c| s + c.to_i}
        ct += 1
    end   
            
    if(nr == 9)
        resp << "#{$_.chomp} is a multiple of 9 and has 9-degree #{ct}"
    else 
        resp << "#{$_.chomp} is not a multiple of 9"
    end
end

resp.each do |s|
  puts s
end

File: rumo9s.py
import sys

for line in sys.stdin:
    if line == "0\n":
        break
        
    line = line.strip()
    org = line

    ct = 0
        
    while int(line) >= 9:
        ct = ct + 1

        s = 0
        t = len(line) 

        for i in range(0,t):
            s = s + int(line[i])

        line = str(s)                
        if(s <= 9):
            break
   
    if(line == '9'):
        print(org," a multiple of 9 and has 9-degree ", ct)
    else:
        print(org, " is not a multiple of 9")
